.travis.yml

```diff
@@ -1,12 +1,31 @@
-script:
-    - docker build --target test --tag my-test-image .
-    - docker run --env-file .env.test my-test-image tests
-    - docker run --env SECRET_KEY --env TRELLO_KEY --env TRELLO_TOKEN my-test-image tests_e2e
+jobs:
```

Comment:
As mentioned below, sadly it looks like we'll need to transition this to Github Actions. I'm more than happy to leave that with you, but I don't want to invent work since you already have this basically working, so in the spoiler tag below is what should be a working solution (though untested!) based on what you've got here; do reach out if it's broken and you're not sure why

<details><summary> Click for yml </summary>

Place the following yml file under: `./github/workflows/<filename>.yml` and check it works as you expect!

```yml
name: To Do app CI & CD
on: [push, pull_request]`

jobs:
  test:
    name: build docker test image and run tests
    runs-on: ubuntu-latest
    env:
        SECRET_KEY: ${{ secrets.SECRET_KEY }}
        TRELLO_KEY: ${{ secrets.TRELLO_KEY }}
        TRELLO_TOKEN: ${{ secrets.TRELLO_TOKEN }}
    steps:
    - uses: actions/checkout@v2
    - run: docker build --target test --tag my-test-image .
    - run: docker run --env-file .env.test my-test-image tests
    - run: docker run --env SECRET_KEY --env TRELLO_KEY --env TRELLO_TOKEN my-test-image tests_e2e

  push:
    name: build docker production image and push to docker hub
    runs-on: ubuntu-latest
    needs: test
    env:
        DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
        DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
    steps:
    - uses: actions/checkout@v2
    - run: echo "$DOCKER_HUB_PASSWORD" | docker login -u "$DOCKER_HUB_USERNAME" --password-stdin
    - run: docker build --target production --tag $DOCKER_HUB_USERNAME/todo:${{ github.sha }} .
    - run: docker push $DOCKER_HUB_USERNAME/todo:${{ github.sha }}

  deploy:
    name: deploy production image to heroku
    runs-on: ubuntu-latest
    env:
        DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
        DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
        HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
    needs: push
    # if: github.ref == 'refs/heads/master' && github.event_name != 'pull_request' # include this line if you want to limit the deployment to the master branch
    steps:
      - uses: actions/checkout@v2
      - run: docker login --username=_ --password=$(heroku auth:token) registry.heroku.com
      - run: docker pull $DOCKER_HUB_USERNAME/todo:${{ github.sha }}
      - run: docker tag $DOCKER_HUB_USERNAME/todo:${{ github.sha }} registry.heroku.com/todo-pdm/web
      - run: docker push registry.heroku.com/todo-pdm/web
      - run: heroku container:release web -a todo-pdm
```
</details>

Reply:
Many thanks for this @JackMead . One small correction needed and got it working.  Deployed in Heroku successfully.